### ISSUE TITLE
TensorFlow backend

### ISSUE THREAD

**Issue by henryiii** [scikit-hep/vector#3]
Work on TensorFlow backend.

**Comment by mayou36** [comment]
So finally TF also makes a step into the right direction by having essentially a (limited) numpy API available as well.
TF and PyTorch should therefore have basically the same implementation and may differ from numpy as the former do not allow(TF)/highly discourage(PyTorch) the usage of inplace operations (or assignments to an existing tensor like in Numpy).

The question is somewhat about numpy (and numba): do we expect to have inplace operations in the numpy implementation actually? If not TF, PyTorch and np can be implemented in one way, otherwise TF and PyTorch may need a separate, yet common to both, implementation.

**Comment by henryiii** [comment]
Great! I think the current plan is to avoid in-place operations. Possibly there could be some room for it in the future, but not something we will target in version 1, and maybe never, since the best way to optimize may be to use numba/tf/pytorch.

I'm doing some quick work for the Princeton flowmeter project, then I'll finish a fairly large commit including more functionality and updates.

**Comment by henryiii** [comment]
(I'm following the Uproot/awkward deadlines, so an initial release is targeting Dec 1)

**Comment by mayou36** [comment]
Great, that's good to hear!Looking forward to it; I'll gladly add the PyTorch/TF backends then

**Comment by eduardo-rodrigues** [comment]
Yep, looking forward to seeing what you have been preparing there :-).

**Comment by henryiii** [comment]
The Protocol in [src/vector/protocols/lorentz.py](https://github.com/scikit-hep/vector/blob/master/src/vector/protocols/lorentz.py) should help with this, as well as the parametrized tests in [tests/test_simple_param.py](https://github.com/scikit-hep/vector/blob/05d1a1dcbe59f0443e0e6bc875b2f10c6801912c/tests/test_simple_param.py) (static link, will likely rename soon).
